include/GafferScene/Shader.h

```diff
@@ -121,6 +121,9 @@ class Shader : public Gaffer::DependencyNode
 		/// \deprecated
 		IECore::ConstObjectVectorPtr state() const;
 
+		virtual void attributesHash( IECore::MurmurHash &h, const Gaffer::Plug *output ) const;
```

Comment:
Was unable to make this protected as it's required by the ShaderPlug but I propose we move Shader::NetworkBuilder to ShaderPlug.